### ISSUE TITLE
wrong AX line number and line range (location and length) on macOS using the macOS Accessibility Framework

### ISSUE THREAD

**Issue by godbout** [electron/electron#47393]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

34.0.0

### What operating system(s) are you using?

macOS

### Operating System Version

macOS Sequoia 15.5 (24F74)

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

using the macOS Accessibility Framework:
1. getting the line number should return 0 if on first line, 1 if on second line, 2 if on third line, etc.
2. getting the line range should return the correct location of the beginning of the line, and the correct line length

### Actual Behavior

1. the line number is only correct if the caret location is at the beginning of a line. if it is anywhere else on that line, the line number is wrong (ends up being the line number of the NEXT line)
2. the line range is also only correct if the caret location is at the beginning of a line. anywhere else and what we get is the data from the NEXT line

### Additional Information

a video showcasing the issue:

https://github.com/user-attachments/assets/aa60283a-9d3d-468c-8bb5-71e0a5463057

also a project that you can run to test: https://github.com/godbout/AXRangeForLineElectronBug

this happens in any app that is built on top of Electron. makes it impossible to use the macOS AX Framework to do cool stuff on Electron apps.

**Comment by codebytere** [comment]
@godbout does this also happen with Chrome?

**Comment by godbout** [comment]
> [@godbout](https://github.com/godbout) does this also happen with Chrome?

Chrome has its own set of issues. for example on text fields only the line number is wrong. line range is correct, wherever your caret location is. for text areas both line number or line range are incorrect, although the line range issue is different from what i see with Electron. (Chrome it always gives the data from the first line, Electron gives the data from the next line.)

**Comment by nikwen** [comment]
Thanks for reporting!

Even though you say it affects all Electron apps, could you please use [Electron Fiddle](https://www.electronjs.org/fiddle) to create a simple example app and publish it from Electron Fiddle as a GitHub gist? Applying the `blocked/need-repro` label until then.

**Comment by godbout** [comment]
honestly fixing this would bring such a huge potential to Electron apps. regarding Chrome/Chromium i've filled a couple of issues years ago but nothing has been done.

**Comment by godbout** [comment]
> Thanks for reporting!
> 
> Even though you say it affects all Electron apps, could you please use [Electron Fiddle](https://www.electronjs.org/fiddle) to create a simple example app and publish it from Electron Fiddle as a GitHub gist? Applying the `blocked/need-repro` label until then.

alright, will try. thanks.

**Comment by godbout** [comment]
@nikwen here you go: https://gist.github.com/godbout/a0cf7b46e4032790bb22281f9cdb39a4

i've used v36.4.0, and although the behavior is still not the one expected, it's now what i'm describing above regarding Chrome. which means the line number is not correct if not on the first character of a line, and line range returned is always the line range from the first line (which, in case of text fields, therefore happens to be correct lol).

thanks for having a look at this, and please lemme know if this should be reported to Chrome/Chromium/Blink also/instead. (although it will affect every Electron app, so i think if it's a Chrome/Chromium/Blink issue you guys should really push for it to be fixed!)

**Comment by nikwen** [comment]
> i've used v36.4.0, and although the behavior is still not the one expected, it's now what i'm describing above regarding Chrome.

Thanks for making the gist and testing on v36!

If the behavior is the same as in Chrome, that means that Electron isn't doing anything special here and just inheriting Chromium's behavior. In those cases, there's no Electron-specific code, so the issue needs to be fixed and reported in Chromium.

Please file an issue for this bug in the [Chromium issue tracker](https://issues.chromium.org/issues). Please include the `index.html` file from the your testcase gist. That will help Chromium developers reproduce the issue (and thus it will increase the chances of it being fixed).

Then please share the link to the Chromium issue here.

Because this is an upstream issue, I'll close this issue in the Electron issue tracker. However, as mentioned, please do report it in the Chromium bug tracker to get it fixed. Thanks!

**Comment by godbout** [comment]
as a starter, issue regarding the line number that i reported in 2022: https://issues.chromium.org/issues/40839954

i'll ping them again regarding this. will open a new issue regarding the line range, and will update this issue with a link to the new bug report.

**Comment by godbout** [comment]
new showcase GH repo to make it clear it's a Chromium bug, not an Electron one per se: https://github.com/godbout/AXLineNumberAndLineRangeChromiumBugs

**Comment by godbout** [comment]
new Chromium issue: https://issues.chromium.org/issues/423048796. now time to thoughts and prayers.
